Time state

jidicula opened the pull request:
Tracking time state with a generator. Currently, the generator is advanced when `apply_action()` is called.

jidicula committed:
Add correct call to time generator

Hooked into the `apply_action()` method and added an `append()` for
the applied action to `self.action_history`

Signed-off-by: jidicula <johanan.idicula@gmail.com>

jidicula committed:
squash! Merge pull request #6 from pseudo-apt/marie

Create generator for advancing time

Signed-off-by: jidicula <johanan.idicula@gmail.com>

# Conflicts:
#	Main.py

davidlougheed commented on internals/__init__.py:
```diff
@@ -53,6 +53,13 @@ def __init__(self, delta_energy, delta_fulfillment):
 }
 
 
+def advance_time():
```
maybe rename this, since its a generator not a function which advances the time

jidicula replied:
Addressed in dc42261

davidlougheed commented on internals/__init__.py:
```diff
@@ -61,11 +68,16 @@ class QuarantineStatus(object):
         self.energy += action.delta_energy
         self.fulfillment += action.delta_fulfillment
+        self.current_time = next(self.time_gen)
+        self.action_history.append(action)
```
this stuff will have to get reconciled with my PR but other than my one req lgtm

jidicula committed:
Rename time generator

https://github.com/pseudo-apt/quarantine-fourteen/pull/5#pullrequestreview-442636242

Signed-off-by: jidicula <johanan.idicula@gmail.com>
